slam-libraries/viam-orb-slam3/orbslam_server_v1.h

```diff
@@ -94,6 +104,7 @@ class SLAMServiceImpl final : public SLAMService::Service {
     std::atomic<bool> finished_processing_offline{false};
     std::thread *thread_save_atlas_as_osa_with_timestamp;
 
+    ORB_SLAM3::System *slam;
```

Comment:
Please let me know if I should be using unique pointer or a different data type for this.

Reply:
I think we would want a unique pointer if possible, as that is the idiomatic(am I using that word right) way to handle this for c++

Reply:
We can only have one unique pointer per raw pointer, so you should either leave this as a raw pointer or move ownership of the raw pointer here in main().

Reply:
Got it. I spent an hour trying out refactoring the code to make it so that there is only one unique pointer to slam & using that pointer both for the data threads & for the ArchiveSlam method however, it ended up being quite a large change (one which I would love to do) but I get the impression we are trying to defer any large refactors until modularization.

Is leaving this as is sufficiently safe?